Mathlib/LinearAlgebra/PiTensorProduct.lean

```diff
@@ -463,17 +471,99 @@ then there is an induced map `⨂ᵢ sᵢ → ⨂ᵢ tᵢ` by `⨂ aᵢ ↦ ⨂
 
 This is `TensorProduct.map` for an arbitrary family of modules.
 -/
-def map (f : Π i, s i →ₗ[R] t i) : (⨂[R] i, s i) →ₗ[R] ⨂[R] i, t i :=
+def map : (⨂[R] i, s i) →ₗ[R] ⨂[R] i, t i :=
   lift <| (tprod R).compLinearMap f
 
-@[simp] lemma map_tprod (f : Π i, s i →ₗ[R] t i) (x : Π i, s i) :
+@[simp] lemma map_tprod (x : Π i, s i) :
     map f (tprod R x) = tprod R fun i ↦ f i (x i) :=
   lift.tprod _
 
+-- No lemmas about associativity, because we don't have associativity of `PiTensorProduct` yet.
+
+theorem map_range_eq_span_tprod : LinearMap.range (map f) =
+    Submodule.span R { t | ∃ (m : (i : ι) → s i), tprod R (fun i ↦ f i (m i)) = t } := by
+  simp only [← Submodule.map_top, ← span_tprod_eq_top, Submodule.map_span]
+  congr; ext t
+  simp only [Set.mem_image, Set.mem_setOf_eq, exists_exists_eq_and, map_tprod]
+
+/-- Given submodules `p i ⊆ s i`, this is the natural map: `⨂[R] i, p i → ⨂[R] i, s i`. -/
+@[simp]
+def mapIncl (p : (i : ι) → Submodule R (s i)) : (⨂[R] i, p i) →ₗ[R] ⨂[R] i, s i :=
+  map (fun (i : ι) ↦ (p i).subtype)
```

Comment:
I'm not convinced this is necessarily worth giving a name, but another reviewer can tie-break

Reply:
I only gave it a name because of the existence of  [`TensorProduct.mapIncl`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/TensorProduct.html#TensorProduct.mapIncl),

Reply:
Ah, that's a good enough reason; can you reference that in the docstring?

Reply:
Done.